tests/snapshot_tests/test_snapshots.py

```diff
@@ -1488,3 +1489,41 @@ def test_scroll_page_down(snap_compare):
     assert snap_compare(
         SNAPSHOT_APPS_DIR / "scroll_page.py", press=["pagedown"], terminal_size=(80, 25)
     )
+
+
+def test_maximize(snap_compare):
+    """Check that maximize isolates a single widget."""
+
+    class MaximizeApp(App):
+        BINDINGS = [("m", "screen.maximize", "maximize focused widget")]
+
+        def compose(self) -> ComposeResult:
+            yield Button("Hello")
+            yield Button("World")
+            yield Footer()
+
+    assert snap_compare(MaximizeApp(), press=["m"])
+
+
+def test_maximize_container(snap_compare):
```

Comment:
I'd suggest you rename this to `test_maximize_widget_in_container` and then add another test that maximizes a container. From looking at the code I'm not 100% sure that's going to work as expected.

Reply:
There are two tests, one that maximizes a widget, and one that maximizes a widget in a container. I think the covers the two cases?